public.tf

```diff
@@ -22,12 +22,12 @@ locals {
 }
 
 resource "aws_subnet" "public" {
-  count             = "${length(var.availability_zones)}"
-  vpc_id            = "${data.aws_vpc.default.id}"
-  availability_zone = "${element(var.availability_zones, count.index)}"
-  cidr_block        = "${cidrsubnet(signum(length(var.cidr_block)) == 1 ? var.cidr_block : data.aws_vpc.default.cidr_block, ceil(log(local.public_subnet_count * 2, 2)), local.public_subnet_count + count.index)}"
-
-  tags = "${merge(module.public_subnet_label.tags, map("Name",format("%s%s%s", module.public_subnet_label.id, var.delimiter, replace(element(var.availability_zones, count.index),"-",var.delimiter))))}"
+  count                   = "${length(var.availability_zones)}"
+  vpc_id                  = "${data.aws_vpc.default.id}"
+  availability_zone       = "${element(var.availability_zones, count.index)}"
+  cidr_block              = "${cidrsubnet(signum(length(var.cidr_block)) == 1 ? var.cidr_block : data.aws_vpc.default.cidr_block, ceil(log(local.public_subnet_count * 2, 2)), local.public_subnet_count + count.index)}"
+  map_public_ip_on_launch = "${var.map_public_ip_on_launch}"
```

Comment:
```suggestion
  map_public_ip_on_launch = "${local.map_public_ip_on_launch}"
```

```
locals {
  map_public_ip_on_launch = "${var.map_public_ip_on_launch == "true" ? true : false}"
}
```

Reply:
similar to what we have for `nat_gateway_enabled` (using locals to convert from input strings to booleans or integers)

Reply:
Great thanks. 
So I read that as:
- if it is equal to string "true" set local to bool true else bool false.

Reply:
yes exactly
Terraform recommends using strings instead of boolean for variables because of inconsistent conversion string-to-bool in TF and on the command line.
So use strings in vars, then use locals to convert to bool if other TF resources accept bool